components/payments/cmd/connectors/internal/connectors/atlar/task_initiate_payment.go

```diff
@@ -0,0 +1,223 @@
+	return func(
+		ctx context.Context,
+		logger logging.Logger,
+		connectorID models.ConnectorID,
+		resolver task.StateResolver,
+		scheduler task.Scheduler,
+		storageReader storage.Reader,
+		ingester ingestion.Ingester,
+		metricsRegistry metrics.MetricsRegistry,
+	) error {
+		logger.Info("initiate payment for transfer-initiation %s", transferID)
+
+		transferInitiationID := models.MustTransferInitiationIDFromString(transferID)
+		transfer, err := getTransfer(ctx, storageReader, transferInitiationID, true)
+		if err != nil {
+			return err
+		}
+
+		attrs := metric.WithAttributes(connectorAttrs...)
+		var paymentID *models.PaymentID
+		defer func() {
+			if err != nil {
+				ctx, cancel := contextutil.Detached(ctx)
+				defer cancel()
+				metricsRegistry.ConnectorObjectsErrors().Add(ctx, 1, attrs)
+				if err := ingester.UpdateTransferInitiationPaymentsStatus(ctx, transfer, paymentID, models.TransferInitiationStatusFailed, err.Error(), transfer.Attempts, time.Now()); err != nil {
+					logger.Error("failed to update transfer initiation status: %v", err)
+				}
+			}
+		}()
+
+		err = ingester.UpdateTransferInitiationPaymentsStatus(ctx, transfer, paymentID, models.TransferInitiationStatusProcessing, "", transfer.Attempts, time.Now())
+		if err != nil {
+			return err
+		}
+
+		attrs = initiatePayoutAttrs
+
+		logger.Info("initiate payment between", transfer.SourceAccountID, " and %s", transfer.DestinationAccountID)
+
+		now := time.Now()
+		defer func() {
+			metricsRegistry.ConnectorObjectsLatency().Record(ctx, time.Since(now).Milliseconds(), attrs)
+		}()
+
+		if transfer.SourceAccount != nil {
+			if transfer.SourceAccount.Type == models.AccountTypeExternal {
+				err = errors.New("payin not implemented: source account must be an internal account")
+				return err
+			}
+		}
+
+		currency, precision, err := currency.GetCurrencyAndPrecisionFromAsset(supportedCurrenciesWithDecimal, transfer.Asset)
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+
+		paymentSchemeType := "SCT" // SEPA Credit Transfer
+		remittanceInformationType := "UNSTRUCTURED"
+		remittanceInformationValue := transfer.Description
+		amount := atlar_models.AmountInput{
+			Currency:    &currency,
+			Value:       transfer.Amount.Int64(),
+			StringValue: amountToString(*transfer.Amount, precision),
+		}
+		date := transfer.ScheduledAt
+		if date.IsZero() {
+			date = time.Now()
+		}
+		dateString := date.Format(time.DateOnly)
+		logger.WithContext(ctx).Infof("date is %s", date.Format(time.RFC3339Nano))
+
+		createPaymentRequest := atlar_models.CreatePaymentRequest{
+			SourceAccountID:              &transfer.SourceAccount.Reference,
+			DestinationExternalAccountID: &transfer.DestinationAccount.Reference,
+			Amount:                       &amount,
+			Date:                         &dateString,
+			ExternalID:                   serializeAtlarPaymentExternalID(transfer.ID.Reference, transfer.Attempts),
```

Comment:
I forgot to mention that the reason I'm not using `transferInitiationID`, but `transfer.ID.Reference` instead is, that Atlar has a character limit on the `externalID`. The limit is not long enough to fit the `transferInitiationID`.

Reply:
Reference is fine also !